### ISSUE TITLE
Improving training performance - Question

### ISSUE THREAD

**Issue by juangea** [SWivid/F5-TTS#379]
I have some questions about trainig because I don't fully understand the parameters.

In my GPU I have 24Gb of Vram, but I see how only 12.4 Gb are being used, my dataset is around 9hours, and so far it's looking good, however I'm wondering how to leverage those extra 12Gb (or 10Gb to leave some room) to improve training performance, this are my settings, how can I improve performance and maximize my hardware usage in the finetuning?

Thanks!

![image](https://github.com/user-attachments/assets/eaed34d2-f8ac-4c09-abf1-469eeffaac56)

**Comment by lpscr** [comment]
@juangea change the Batch Size Per gpu to `10000` this you get about 20gb See what works better: adding more GPU memory you going out-of-memory this about the lenght audio you use

**Comment by juangea** [comment]
Great, I tried incrementing that to 6000 and it reached 18Gb, so I will try incrementing to 10000 next time.

Will this improve training speed or will affect training quality?

Also the rest of the parameters, like "Gradient Accumulation Steps or Max Gradient Norm, or even learning rate, what are they for?  could be useful to touch them or they are ok as they are?

And a final question, I'm training for F5, could it be good to train also for E2, is there any actual difference/leverage?

Thanks for the info!

**Comment by SWivid** [comment]
> Batchsize, Gradient Accumulation Steps, Max Gradient Norm, Learning Rate, etc.

These are hyperparam for neural network training, simply search and will have blogs for these terms e.g.:
https://towardsdatascience.com/what-is-gradient-accumulation-in-deep-learning-ec034122cfa

> And a final question, I'm training for F5, could it be good to train also for E2, is there any actual difference/leverage?

@juangea our paper is actually for that usage, fully detailed explanations there covering all questions ~ Mainly model structure difference and additional sampling strategy

**Comment by juangea** [comment]
Awesome, thanks @SWivid ! :)
